src/mlpro/bf/math/normalizers.py

```diff
@@ -0,0 +1,319 @@
+ ## -------------------------------------------------------------------------------------------------
+## -- Project : MLPro - A Synoptic Framework for Standardized Machine Learning Tasks
+## -- Package : mlpro.bf.pool
+## -- Module  : normalizers.py
+## -------------------------------------------------------------------------------------------------
+## -- History :
+## -- yyyy-mm-dd  Ver.      Auth.    Description
+## -- 2022-09-16  0.0.0     LSB      Creation
+## -- 2022-09-20  1.0.0     LSB      Release of first version
+## -- 2022-09-23  1.0.1     LSB      Refactoring
+## -- 2022-09-26  1.0.2     LSB      Refatoring and reduced custom normalize and denormalize methods
+## -------------------------------------------------------------------------------------------------
+
+"""
+Ver. 1.0.2 (2022-09-25)
+This module provides base class for Normalizers and normalizer objects including MinMax normalization and 
+normalization by Z transformation.
+"""
+
+from mlpro.bf.math import *
+from mlpro.rl.models_sar import *
+import numpy as np
+from typing import Union
+
+
+
+## -------------------------------------------------------------------------------------------------
+## -------------------------------------------------------------------------------------------------
+class Normalizer:
+    """
+    Base template class for normalizer objects
+    """
+    C_TYPE = 'Normalizer'
+    C_NAME = ''
+
+
+## -------------------------------------------------------------------------------------------------
+    def __init__(self):
+
+        self._param = None
+        self._param_old = None
+
+
+## -------------------------------------------------------------------------------------------------
+    def get_parameters(self):
+        """
+        method to get the normalization parameters
+        """
+        return self._param
+
+
+## -------------------------------------------------------------------------------------------------
+    def set_parameters(self, p_data):
```

Comment:
Please make this a protected method. It is used internally to switch between old and current parameter sets. Parameter is not data but parameters and it can be implemented here:

self._param = p_param